uportal-war/src/main/java/org/jasig/portal/portlets/dynamicskin/FileSystemDynamicSkinService.java

```diff
@@ -249,4 +250,33 @@ public String calculateTokenForCurrentSkin(PortletRequest request) {
         return Integer.toString(hash);
     }
 
+    @Override
+    public SortedSet<String> getSkinNames(PortletRequest request) {
+        // Context to access the filesystem
+        PortletContext ctx = request.getPortletSession().getPortletContext();
+
+        // Determine the full path to the skins directory
+        String skinsFilepath = ctx.getRealPath(rootFolder);
+
+        // Create File object to access the filesystem
+        File skinsDir = new File(skinsFilepath);
```

Comment:
@drewwills BTW I assumed the skin-key was the filename and I returned that.  I assumed the skin-name was the friendly user name.  If that is not the correct convention, let me know but it looked like the javascript code was honoring that convention.